sys/include/crypto/padding.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2017 Mathias Tausig
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_crypto
+ * @{
+ *
+ * @file        padding.h
+ * @brief       Padding modes for encryption
+ *
+ * @author      Mathias Tausig
+ */
+
+#ifndef CRYPTO_PADDING_H
+#define CRYPTO_PADDING_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include "crypto/ciphers.h"
+
+/**
+ * @brief   PKCS#7 type value
+ */
+#define PADDING_TYPE_PKCS7 1
+
+/**
+ * @brief   The output buffer is too small
+ */
+#define PADDING_NOT_ENOUGH_SPACE_AVAILABLE -1
+
+/**
+ * @brief   Invalid blocksize
+ */
+#define PADDING_INVALID_BLOCKSIZE -2
+
+/**
+ * @brief   The padding is invalid
+ */
+#define PADDING_INVALID_PADDING -3
+
+/**
+ * @brief   The input data is larger than INT32MAX
+ */
+#define PADDING_DATA_TOO_LONG -4
+
+/**
+ * @brief Apply PKCS#7 padding to some date
+ *
+ * @param data              The data to be padded
+ * @param data_len          The length of the data
+ * @param blocksize         The length of the block. The padded data will be a multiple of this value.
+ * @param padded_data       Place to store the padded data
```

Comment:
So this will copy all bytes from `data` to `padded_data` with additional padding?
So `padded_data_len` must be > than `data_len`?

Reply:
Is there also an option if the buffer is already large enough and I just want to fill out the remaining bytes with padding, without copying it to a second buffer?

Reply:
this file does not exist anymore